WordPress/Classes/Services/BlogService.m

```diff
@@ -744,6 +734,27 @@ - (NSTimeZone *)timeZoneForBlog:(Blog *)blog
     return timeZone;
 }
 
+- (void)updateBlog:(Blog *)blog settingsWithRemoteSettings:(RemoteBlogSettings *)remoteSettings
+{
+    blog.blogName = remoteSettings.name;
+    blog.blogTagline = remoteSettings.desc;
+    if (remoteSettings.defaultCategory) {
```

Comment:
When accessing self-hosted sites these values are not returned so in remoteSettings they are nil, so that is why they are being check.